api/src/core/utils.ts

```diff
@@ -66,3 +66,8 @@ export const flatDeep = (arr, d = 1) => {
   return d > 0 ? arr.reduce((acc, val) => acc.concat(Array.isArray(val) ? flatDeep(val, d - 1) : val), [])
                 : arr.slice();
 };
+
+// Partial type with nested properties also as Partial.
+export type RecursivePartial<T> = {
+    [P in keyof T]?: RecursivePartial<T[P]>;
+};
```

Comment:
Why is this needed?

Reply:
In the new test that when supplying the mock data with type `Project` entity, I don't want to code many of unnecessary fields (just id, name and forestClient), so I used typescript `Partial<Project>`. However, this is not in deep level, so it still had error for the type for something like `project.forestClient` relationship field when building a mock data. So, this piece will make it a deep `Partial`.

Reply:
Sounds good.